### PR TITLE
[I4Y3TQ] Fix Polling Workers Sync Issue For Resume Flow

### PR DESCRIPTION
**Pull request by sumanth43**
### What type of PR is this?

Uncomment only one /kind <> line, hit enter to put that in a new line, and remove leading whitespaces from that line:

kind bug 

### Which issue(s) this PR fixes:

Fixes #I4Y3TQ

### Special notes for your reviewers:

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/remotetask/HttpRemoteTask.java [hunk 1]
```diff
@@ -644,7 +644,7 @@ public synchronized void cancelToResume()
 
     private void sendCancelRequest(TaskStatus taskStatus, TaskState targetState, String action)
     {
-        log.debug("Cancelling task %s, with target state %s", taskStatus.getTaskId(), targetState);
+        log.info("Cancelling task %s, with target state %s", taskStatus.getTaskId(), targetState);
```
Comment (sumanth43) [inline presto-main/src/main/java/io/prestosql/server/remotetask/HttpRemoteTask.java:647]: will revert this change

Comment (sumanth43) [reply presto-main/src/main/java/io/prestosql/server/remotetask/HttpRemoteTask.java]: reverted

**Comment by Nitin-Kashyap** [comment]
lgtm

File: presto-main/src/main/java/io/prestosql/failuredetector/HeartbeatFailureDetector.java [hunk 2]
```diff
@@ -221,6 +222,75 @@ public int getActiveCount()
...
+        // this while ensures for one loop of updating all tasks or any failed task is identified
+        while (!(newResponseTime > oldResponseTime || getFailed().size() > 0)) {
+            newStats = getStats();
+            newResponseTime = newStats.values().stream().mapToLong(stats -> stats.getLastResponseTime().getMillis()).min().getAsLong();
+        }
+        double newFailuresCount = newStats.values().stream().mapToDouble(stats -> stats.getRecentFailures()).sum();
+        // this while ensures for updating tasks till
+        // 1. Either receiving failure responses is stopped
+        // 2. Failure responses reached the configured failureRatioThreshold
+        while (newFailuresCount > oldFailuresCount) {
```
Comment (sraghunandan) [inline presto-main/src/main/java/io/prestosql/failuredetector/HeartbeatFailureDetector.java:259]: cant we use do..while and complete the duplicate work in single loop?

File: presto-main/src/main/java/io/prestosql/failuredetector/HeartbeatFailureDetector.java [hunk 3]
```diff
@@ -221,6 +222,75 @@ public int getActiveCount()
         return builder.build();
     }
 
+    @Override
+    public void updateMonitorServices()
```
Comment (sraghunandan) [inline presto-main/src/main/java/io/prestosql/failuredetector/HeartbeatFailureDetector.java:226]: refactor updateMonitoredServices() so that the common functionality is not duplicated

**Comment by Nitin-Kashyap** [comment]
lgtm
